modules/videoio/src/cap_gstreamer.cpp

```diff
@@ -946,6 +954,7 @@ bool GStreamerCapture::retrieveVideoFrame(int, OutputArray dst)
     if (!buf)
         return false;
 
+    timestamp = GST_BUFFER_PTS(buf);
```

Comment:
the timestamp has been updated on grabXXXFrame. Looks like it's duplicate and may be removed.

Reply:
Done